libs/experimental/langchain_experimental/graph_transformers/llm.py

```diff
@@ -158,6 +159,90 @@ def map_to_base_relationship(rel: Any) -> Relationship:
+    if not raw_schema["parsed"]:
+        try:
+            argument_json = json.loads(
+                raw_schema["raw"].additional_kwargs["tool_calls"][0]["function"][
+                    "arguments"
+                ]
+            )
+            nodes = []
+            for node in argument_json["nodes"]:
+                if not node.get("id"):  # Id is mandatory
+                    continue
+                nodes.append(
+                    Node(
+                        id=node["id"].title(),
+                        type=node.get("type", "Node").capitalize(),
+                    )
+                )
+            relationships = []
+            for rel in argument_json["relationships"]:
+                # Mandatory props
+                if (
+                    not rel.get("source_node_id")
+                    or not rel.get("target_node_id")
+                    or not rel.get("type")
+                ):
+                    continue
+
+                # Node type copying if needed from node list
+                if not rel.get("source_node_type"):
+                    try:
+                        rel["source_node_type"] = [
+                            el["type"]
+                            for el in argument_json["nodes"]
+                            if el["id"] == rel["source_node_id"]
+                        ][0]
+                    except IndexError:
+                        rel["source_node_type"] = "Node"
+                if not rel.get("target_node_type"):
+                    try:
+                        rel["target_node_type"] = [
+                            el["type"]
+                            for el in argument_json["nodes"]
+                            if el["id"] == rel["target_node_id"]
+                        ][0]
+                    except IndexError:
+                        rel["target_node_type"] = "Node"
+
+                source_node = Node(
+                    id=rel["source_node_id"].title(),
+                    type=rel["source_node_type"].capitalize(),
+                )
+                target_node = Node(
+                    id=rel["target_node_id"].title(),
+                    type=rel["target_node_type"].capitalize(),
+                )
+                relationships.append(
+                    Relationship(
+                        source=source_node,
+                        target=target_node,
+                        type=rel["type"].replace(" ", "_").upper(),
+                    )
+                )
+        except Exception:  # If we can't parse JSON
+            return [], []
+    else:  # If there are no validation errors use parsed pydantic object
+        raw_schema = raw_schema["parsed"]
+        nodes = (
+            [map_to_base_node(node) for node in raw_schema.nodes]
+            if raw_schema.nodes
+            else []
+        )
+
+        relationships = (
+            [map_to_base_relationship(rel) for rel in raw_schema.relationships]
```

Comment:
is the .title() / .capitalize()  / upper() happening here?

Reply:
yes